broker/broker_test.go

```diff
@@ -464,13 +466,137 @@ var _ = Describe("Broker", func() {
 			}))
 		})
 
+		It("updates the redis replication group through the Provider", func() {
+			validUpdateDetails.RawParameters = []byte(`{"preferred_maintenance_window": "mon:23:00-tue:01:30"}`)
+
+			spec, err := b.Update(context.Background(), "instanceid", validUpdateDetails, true)
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(fakeProvider.UpdateReplicationGroupCallCount()).To(Equal(1))
+			_, id, params := fakeProvider.UpdateReplicationGroupArgsForCall(0)
+
+			Expect(id).To(Equal("instanceid"))
+
+			expectedParameters := providers.UpdateReplicationGroupParameters{
+				PreferredMaintenanceWindow: "mon:23:00-tue:01:30",
+			}
+
+			Expect(params).To(Equal(expectedParameters))
+
+			Expect(spec).To(Equal(brokerapi.UpdateServiceSpec{
+				IsAsync:       true,
+				OperationData: broker.Operation{Action: broker.ActionUpdating}.String(),
+			}))
```

Comment:
Is it worth also asserting that there are zero calls to `fakeProvider.UpdateParams()` in this case?

Reply:
yep, done